File: app/src/main/java/com/android/mathias/velocity/DBManager.java
package com.android.mathias.velocity;


import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.provider.BaseColumns;

import com.google.android.gms.maps.model.LatLng;

import java.text.DateFormat;
import java.text.ParseException;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

final class DBManager {

    private DBManager() {}

    static void saveWalk(Context context, Walk walk) {
        DBHelperWalks dbHelper = new DBHelperWalks(context);
        SQLiteDatabase db = dbHelper.getWritableDatabase();
        ContentValues values = new ContentValues();
        values.put(WalkEntry.COLUMN_NAME_ROUTE, walk.getRoute().getName());
        values.put(WalkEntry.COLUMN_NAME_DURATION, walk.getDuration());
        values.put(WalkEntry.COLUMN_NAME_DATE, DateFormat.getDateTimeInstance(DateFormat.LONG,DateFormat.LONG).format(walk.getDate()));
        long newRowId = db.insert(WalkEntry.TABLE_NAME, null, values);
    }

    static List<Walk> getWalks(Context context, Route route) {
        DBHelperWalks dbHelper = new DBHelperWalks(context);
        SQLiteDatabase db = dbHelper.getReadableDatabase();
        String[] projection = {
            WalkEntry._ID,
            WalkEntry.COLUMN_NAME_ROUTE,
            WalkEntry.COLUMN_NAME_DURATION,
            WalkEntry.COLUMN_NAME_DATE
        };
        String selection = WalkEntry.COLUMN_NAME_ROUTE + " LIKE ?";
        String[] selectionArgs = { route != null ? route.getName() : "%" };
        String sortOrder = WalkEntry.COLUMN_NAME_DATE + " DESC";
        Cursor c = db.query(
            WalkEntry.TABLE_NAME,    // The table to query
            projection,              // The columns to return
            selection,               // The columns for the WHERE clause
            selectionArgs,           // The values for the WHERE clause
            null,                    // don't group the rows
            null,                    // don't filter by row groups
            sortOrder                // The sort order
        );
        List<Walk> walks = new ArrayList<>();
        if (c != null) {
            if (c.moveToFirst()) {
                do {
                    Route walkRoute = new Route(c.getString(c.getColumnIndexOrThrow(WalkEntry.COLUMN_NAME_ROUTE)));
                    long walkDuration = c.getLong(c.getColumnIndexOrThrow(WalkEntry.COLUMN_NAME_DURATION));
                    Date walkDate = new Date();
                    try {
                        DateFormat df = DateFormat.getDateTimeInstance(DateFormat.LONG,DateFormat.LONG);
                        walkDate = df.parse(c.getString(c.getColumnIndexOrThrow(WalkEntry.COLUMN_NAME_DATE)));
                    } catch (ParseException e) { e.printStackTrace(); }
                    walks.add(new Walk(walkDuration, walkDate, walkRoute));
                } while (c.moveToNext());
            }
        }
        c.close();
        return walks;
    }

    public static void deleteWalks(Context context, Date date) {
        DBHelperWalks dbHelper = new DBHelperWalks(context);
        SQLiteDatabase db = dbHelper.getWritableDatabase();
        String selection = WalkEntry.COLUMN_NAME_DATE + " LIKE ?";
        String[] selectionArgs = { date.toString() };
        db.delete(WalkEntry.TABLE_NAME, selection, selectionArgs);
    }
    public static void deleteWalks(Context context, Route route) {
        DBHelperWalks dbHelper = new DBHelperWalks(context);
        SQLiteDatabase db = dbHelper.getWritableDatabase();
        String selection = WalkEntry.COLUMN_NAME_ROUTE + " LIKE ?";
        String[] selectionArgs = { route.getName() };
        db.delete(WalkEntry.TABLE_NAME, selection, selectionArgs);
    }

    public static void deleteAllWalks(Context context) {
        DBHelperWalks dbHelper = new DBHelperWalks(context);
        SQLiteDatabase db = dbHelper.getWritableDatabase();
        db.delete(WalkEntry.TABLE_NAME, null, null);
    }

    public static void saveRoute(Context context, Route route) {
        DBHelperRoutes dbHelper = new DBHelperRoutes(context);
        SQLiteDatabase db = dbHelper.getWritableDatabase();
        ContentValues values = new ContentValues();
        values.put(RouteEntry.COLUMN_NAME_NAME, route.getName());
        values.put(RouteEntry.COLUMN_NAME_START_LAT, route.getStartLoc().latitude);
        values.put(RouteEntry.COLUMN_NAME_START_LNG, route.getStartLoc().longitude);
        values.put(RouteEntry.COLUMN_NAME_END_LAT, route.getEndLoc().latitude);
        values.put(RouteEntry.COLUMN_NAME_END_LNG, route.getEndLoc().longitude);
        values.put(RouteEntry.COLUMN_NAME_START_NAME, route.getStartName());
        values.put(RouteEntry.COLUMN_NAME_END_NAME, route.getEndName());
        long newRowId = db.insert(RouteEntry.TABLE_NAME, null, values);
    }

    public static List<Route> getRoutes (Context context, String name) {
        DBHelperRoutes dbHelper = new DBHelperRoutes(context);
        SQLiteDatabase db = dbHelper.getReadableDatabase();
        String[] projection = {
            RouteEntry._ID,
            RouteEntry.COLUMN_NAME_NAME,
            RouteEntry.COLUMN_NAME_START_LAT,
            RouteEntry.COLUMN_NAME_START_LNG,
            RouteEntry.COLUMN_NAME_END_LAT,
            RouteEntry.COLUMN_NAME_END_LNG,
            RouteEntry.COLUMN_NAME_START_NAME,
            RouteEntry.COLUMN_NAME_END_NAME
        };
        String selection = RouteEntry.COLUMN_NAME_NAME + " LIKE ?";
        String[] selectionArgs = { name != null ? name : "%" };
        String sortOrder = RouteEntry.COLUMN_NAME_NAME + " DESC";
        Cursor c = db.query(
            RouteEntry.TABLE_NAME,   // The table to query
            projection,              // The columns to return
            selection,               // The columns for the WHERE clause
            selectionArgs,           // The values for the WHERE clause
            null,                    // don't group the rows
            null,                    // don't filter by row groups
            sortOrder                // The sort order
        );
        List<Route> routes = new ArrayList<>();
        if (c != null) {
            if (c.moveToFirst()) {
                do {
                    String routeName = c.getString(c.getColumnIndexOrThrow(RouteEntry.COLUMN_NAME_NAME));
                    double routeStartLat = c.getDouble(c.getColumnIndexOrThrow(RouteEntry.COLUMN_NAME_START_LAT));
                    double routeStartLng = c.getDouble(c.getColumnIndexOrThrow(RouteEntry.COLUMN_NAME_START_LNG));
                    double routeEndLat = c.getDouble(c.getColumnIndexOrThrow(RouteEntry.COLUMN_NAME_END_LAT));
                    double routeEndLng = c.getDouble(c.getColumnIndexOrThrow(RouteEntry.COLUMN_NAME_END_LNG));
                    String routeStartName = c.getString((c.getColumnIndexOrThrow(RouteEntry.COLUMN_NAME_START_NAME)));
                    String routeEndName = c.getString((c.getColumnIndexOrThrow(RouteEntry.COLUMN_NAME_END_NAME)));
                    LatLng startLoc = new LatLng(routeStartLat, routeStartLng);
                    LatLng endLoc = new LatLng(routeEndLat,routeEndLng);
                    routes.add(new Route(routeName, startLoc, endLoc, routeStartName, routeEndName));
                } while (c.moveToNext());
            }
        }
        c.close();
        return routes;
    }

    public static void deleteAllRoutes(Context context) {
        DBHelperRoutes dbHelper = new DBHelperRoutes(context);
        SQLiteDatabase db = dbHelper.getWritableDatabase();
        db.delete(RouteEntry.TABLE_NAME, null, null);
    }

    private static final class DBHelperWalks extends SQLiteOpenHelper {
        // METADATA
        private static final int DATABASE_VERSION = 1;
        private static final String DATABASE_NAME = "Walks.db";
        // QUERIES
        private static final String TEXT_TYPE = " TEXT";
        private static final String INTEGER_TYPE = " INTEGER";
        private static final String COMMA_SEP = ",";
        private static final String SQL_DELETE_ENTRIES = "DROP TABLE IF EXISTS " + WalkEntry.TABLE_NAME;
        private static final String SQL_CREATE_ENTRIES =
            "CREATE TABLE " + WalkEntry.TABLE_NAME + " (" +
                    WalkEntry._ID + " INTEGER PRIMARY KEY," +
                    WalkEntry.COLUMN_NAME_ROUTE + TEXT_TYPE + COMMA_SEP +
                    WalkEntry.COLUMN_NAME_DURATION + INTEGER_TYPE +COMMA_SEP +
                    WalkEntry.COLUMN_NAME_DATE + TEXT_TYPE + " )";

        DBHelperWalks(Context context) {
            super(context, DATABASE_NAME, null, DATABASE_VERSION);
        }

        @Override
        public void onCreate(SQLiteDatabase db) {
            db.execSQL(SQL_CREATE_ENTRIES);
        }

        @Override
        public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
            db.execSQL(SQL_DELETE_ENTRIES);
            onCreate(db);
        }

        @Override
        public void onDowngrade(SQLiteDatabase db, int oldVersion, int newVersion) {
            onUpgrade(db, oldVersion, newVersion);
        }
    }

    private static class WalkEntry implements BaseColumns {
        private static final String TABLE_NAME = "walks";
        private static final String COLUMN_NAME_ROUTE = "route";
        private static final String COLUMN_NAME_DURATION = "duration";
        private static final String COLUMN_NAME_DATE = "date";
    }

    private static final class DBHelperRoutes extends SQLiteOpenHelper {
        // METADATA
        private static final int DATABASE_VERSION = 1;
        private static final String DATABASE_NAME = "Routes.db";
        // QUERIES
        private static final String TEXT_TYPE = " TEXT";
        private static final String REAL_TYPE = " REAL";
        private static final String COMMA_SEP = ",";
        private static final String SQL_DELETE_ENTRIES = "DROP TABLE IF EXISTS " + RouteEntry.TABLE_NAME;
        private static final String SQL_CREATE_ENTRIES =
                "CREATE TABLE " + RouteEntry.TABLE_NAME + " (" +
                        RouteEntry._ID + " INTEGER PRIMARY KEY," +
                        RouteEntry.COLUMN_NAME_NAME + TEXT_TYPE + COMMA_SEP +
                        RouteEntry.COLUMN_NAME_START_LAT + REAL_TYPE +COMMA_SEP +
                        RouteEntry.COLUMN_NAME_START_LNG + REAL_TYPE +COMMA_SEP +
                        RouteEntry.COLUMN_NAME_END_LAT + REAL_TYPE +COMMA_SEP +
                        RouteEntry.COLUMN_NAME_END_LNG + REAL_TYPE +COMMA_SEP +
                        RouteEntry.COLUMN_NAME_START_NAME + TEXT_TYPE +COMMA_SEP +
                        RouteEntry.COLUMN_NAME_END_NAME + TEXT_TYPE + " )";

        DBHelperRoutes(Context context) {
            super(context, DATABASE_NAME, null, DATABASE_VERSION);
        }

        @Override
        public void onCreate(SQLiteDatabase db) {
            db.execSQL(SQL_CREATE_ENTRIES);
        }

        @Override
        public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
            db.execSQL(SQL_DELETE_ENTRIES);
            onCreate(db);
        }

        @Override
        public void onDowngrade(SQLiteDatabase db, int oldVersion, int newVersion) {
            onUpgrade(db, oldVersion, newVersion);
        }
    }

    private static class RouteEntry implements BaseColumns {
        private static final String TABLE_NAME = "routes";
        private static final String COLUMN_NAME_NAME = "name";
        private static final String COLUMN_NAME_START_LAT = "start_lat";
        private static final String COLUMN_NAME_START_LNG = "start_lng";
        private static final String COLUMN_NAME_END_LAT = "end_lat";
        private static final String COLUMN_NAME_END_LNG = "end_lng";
        private static final String COLUMN_NAME_START_NAME = "start_name";
        private static final String COLUMN_NAME_END_NAME = "end_name";
    }
}

File: app/src/main/java/com/android/mathias/velocity/DBManager.kt
package com.android.mathias.velocity


import android.content.ContentValues
import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import android.provider.BaseColumns
import com.google.android.gms.maps.model.LatLng
import java.text.DateFormat
import java.text.ParseException
import java.util.*

internal object DBManager {

    fun saveWalk(context: Context, walk: Walk) {
        val dbHelper = DBHelperWalks(context)
        val db = dbHelper.writableDatabase
        val values = ContentValues()
        values.put(WalkEntry.COLUMN_NAME_ROUTE, walk.route!!.name)
        values.put(WalkEntry.COLUMN_NAME_DURATION, walk.duration)
        values.put(WalkEntry.COLUMN_NAME_DATE, DateFormat.getDateTimeInstance(DateFormat.LONG, DateFormat.LONG).format(walk.date))
        db.insert(WalkEntry.TABLE_NAME, null, values)
    }

    fun getWalks(context: Context, route: Route?): List<Walk> {
        val dbHelper = DBHelperWalks(context)
        val db = dbHelper.readableDatabase
        val projection = arrayOf(WalkEntry.ID, WalkEntry.COLUMN_NAME_ROUTE, WalkEntry.COLUMN_NAME_DURATION, WalkEntry.COLUMN_NAME_DATE)
        val selection = WalkEntry.COLUMN_NAME_ROUTE + " LIKE ?"
        val selectionArgs = arrayOf(if (route != null) route.name else "%")
        val sortOrder = WalkEntry.COLUMN_NAME_DATE + " DESC"
        val c = db.query(
                WalkEntry.TABLE_NAME, // The table to query
                projection, // The columns to return
                selection, // The columns for the WHERE clause
                selectionArgs, null, null, // don't filter by row groups
                sortOrder                // The sort order
        )// The values for the WHERE clause
        // don't group the rows
        val walks = ArrayList<Walk>()
        if (c != null) {
            if (c.moveToFirst()) {
                do {
                    val walkId = c.getLong(c.getColumnIndexOrThrow(WalkEntry.ID))
                    val walkRoute = Route(c.getString(c.getColumnIndexOrThrow(WalkEntry.COLUMN_NAME_ROUTE)))
                    val walkDuration = c.getLong(c.getColumnIndexOrThrow(WalkEntry.COLUMN_NAME_DURATION))
                    var walkDate = Date()
                    try {
                        val df = DateFormat.getDateTimeInstance(DateFormat.LONG, DateFormat.LONG)
                        walkDate = df.parse(c.getString(c.getColumnIndexOrThrow(WalkEntry.COLUMN_NAME_DATE)))
                    } catch (e: ParseException) {
                        e.printStackTrace()
                    }

                    walks.add(Walk(walkId, walkDuration, walkDate, walkRoute))
                } while (c.moveToNext())
            }
            c.close()
        }
        return walks
    }

    fun deleteWalk(context: Context, id: Long) {
        val dbHelper = DBHelperWalks(context)
        val db = dbHelper.writableDatabase
        val selection = WalkEntry.ID + " = ?"
        val selectionArgs = arrayOf(id.toString())
        db.delete(WalkEntry.TABLE_NAME, selection, selectionArgs)
    }

    fun deleteWalks(context: Context, date: Date) {
        val dbHelper = DBHelperWalks(context)
        val db = dbHelper.writableDatabase
        val selection = WalkEntry.COLUMN_NAME_DATE + " LIKE ?"
        val selectionArgs = arrayOf(date.toString())
        db.delete(WalkEntry.TABLE_NAME, selection, selectionArgs)
    }

    fun deleteWalks(context: Context, route: Route) {
        val dbHelper = DBHelperWalks(context)
        val db = dbHelper.writableDatabase
        val selection = WalkEntry.COLUMN_NAME_ROUTE + " LIKE ?"
        val selectionArgs = arrayOf(route.name)
        db.delete(WalkEntry.TABLE_NAME, selection, selectionArgs)
    }

    fun deleteAllWalks(context: Context) {
        val dbHelper = DBHelperWalks(context)
        val db = dbHelper.writableDatabase
        db.delete(WalkEntry.TABLE_NAME, null, null)
    }

    fun saveRoute(context: Context, route: Route) {
        val dbHelper = DBHelperRoutes(context)
        val db = dbHelper.writableDatabase
        val values = ContentValues()
        values.put(RouteEntry.COLUMN_NAME_NAME, route.name)
        values.put(RouteEntry.COLUMN_NAME_POS, route.pos)
        values.put(RouteEntry.COLUMN_NAME_START_LAT, route.startLoc!!.latitude)
        values.put(RouteEntry.COLUMN_NAME_START_LNG, route.startLoc!!.longitude)
        values.put(RouteEntry.COLUMN_NAME_END_LAT, route.endLoc!!.latitude)
        values.put(RouteEntry.COLUMN_NAME_END_LNG, route.endLoc!!.longitude)
        values.put(RouteEntry.COLUMN_NAME_START_NAME, route.startName)
        values.put(RouteEntry.COLUMN_NAME_END_NAME, route.endName)
        db.insert(RouteEntry.TABLE_NAME, null, values)
    }

    fun getRoutes(context: Context, name: String?): List<Route> {
        val dbHelper = DBHelperRoutes(context)
        val db = dbHelper.readableDatabase
        val projection = arrayOf(RouteEntry.ID, RouteEntry.COLUMN_NAME_POS, RouteEntry.COLUMN_NAME_NAME, RouteEntry.COLUMN_NAME_START_LAT, RouteEntry.COLUMN_NAME_START_LNG, RouteEntry.COLUMN_NAME_END_LAT, RouteEntry.COLUMN_NAME_END_LNG, RouteEntry.COLUMN_NAME_START_NAME, RouteEntry.COLUMN_NAME_END_NAME)
        val selection = RouteEntry.COLUMN_NAME_NAME + " LIKE ?"
        val selectionArgs = arrayOf(name ?: "%")
        val sortOrder = RouteEntry.COLUMN_NAME_POS + " ASC"
        val c = db.query(
                RouteEntry.TABLE_NAME, // The table to query
                projection, // The columns to return
                selection, // The columns for the WHERE clause
                selectionArgs, null, null, // don't filter by row groups
                sortOrder                // The sort order
        )// The values for the WHERE clause
        // don't group the rows
        val routes = ArrayList<Route>()
        if (c != null) {
            if (c.moveToFirst()) {
                do {
                    val routeId = c.getLong(c.getColumnIndexOrThrow(RouteEntry.ID))
                    val routePos = c.getInt(c.getColumnIndexOrThrow(RouteEntry.COLUMN_NAME_POS))
                    val routeName = c.getString(c.getColumnIndexOrThrow(RouteEntry.COLUMN_NAME_NAME))
                    val routeStartLat = c.getDouble(c.getColumnIndexOrThrow(RouteEntry.COLUMN_NAME_START_LAT))
                    val routeStartLng = c.getDouble(c.getColumnIndexOrThrow(RouteEntry.COLUMN_NAME_START_LNG))
                    val routeEndLat = c.getDouble(c.getColumnIndexOrThrow(RouteEntry.COLUMN_NAME_END_LAT))
                    val routeEndLng = c.getDouble(c.getColumnIndexOrThrow(RouteEntry.COLUMN_NAME_END_LNG))
                    val routeStartName = c.getString(c.getColumnIndexOrThrow(RouteEntry.COLUMN_NAME_START_NAME))
                    val routeEndName = c.getString(c.getColumnIndexOrThrow(RouteEntry.COLUMN_NAME_END_NAME))
                    val routeStartLoc = LatLng(routeStartLat, routeStartLng)
                    val routeEndLoc = LatLng(routeEndLat, routeEndLng)
                    routes.add(Route(routeId, routePos, routeName, routeStartLoc, routeEndLoc, routeStartName, routeEndName))
                } while (c.moveToNext())
            }
            c.close()
        }
        return routes
    }

    fun deleteRoute(context: Context, id: Long) {
        val dbHelper = DBHelperRoutes(context)
        val db = dbHelper.writableDatabase
        val selection = RouteEntry.ID + " = ?"
        val selectionArgs = arrayOf(id.toString())
        db.delete(RouteEntry.TABLE_NAME, selection, selectionArgs)
    }

    fun deleteAllRoutes(context: Context) {
        val dbHelper = DBHelperRoutes(context)
        val db = dbHelper.writableDatabase
        db.delete(RouteEntry.TABLE_NAME, null, null)
    }

    fun setRoutePos(context: Context, id: Long, pos: Int) {
        val dbHelper = DBHelperRoutes(context)
        val db = dbHelper.writableDatabase
        val query = "UPDATE " + RouteEntry.TABLE_NAME +
                " SET " + RouteEntry.COLUMN_NAME_POS + " = " + pos +
                " WHERE " + RouteEntry.ID + " = " + id
        db.execSQL(query)
    }

    private class DBHelperWalks internal constructor(context: Context) : SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {

        override fun onCreate(db: SQLiteDatabase) {
            db.execSQL(SQL_CREATE_ENTRIES)
        }

        override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
            db.execSQL(SQL_DELETE_ENTRIES)
            onCreate(db)
        }

        override fun onDowngrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
            onUpgrade(db, oldVersion, newVersion)
        }

        companion object {
            // METADATA
            private const val DATABASE_VERSION = 1
            private const val DATABASE_NAME = "Walks.db"
            // QUERIES
            private const val TEXT_TYPE = " TEXT"
            private const val INTEGER_TYPE = " INTEGER"
            private const val COMMA_SEP = ","
            private const val SQL_DELETE_ENTRIES = "DROP TABLE IF EXISTS " + WalkEntry.TABLE_NAME
            private const val SQL_CREATE_ENTRIES = "CREATE TABLE " + WalkEntry.TABLE_NAME + " (" +
                    WalkEntry.ID + " INTEGER PRIMARY KEY," +
                    WalkEntry.COLUMN_NAME_ROUTE + TEXT_TYPE + COMMA_SEP +
                    WalkEntry.COLUMN_NAME_DURATION + INTEGER_TYPE + COMMA_SEP +
                    WalkEntry.COLUMN_NAME_DATE + TEXT_TYPE + " )"
        }
    }

    private class WalkEntry : BaseColumns {
        companion object {
            const val ID = BaseColumns._ID
            const val TABLE_NAME = "walks"
            const val COLUMN_NAME_ROUTE = "route"
            const val COLUMN_NAME_DURATION = "duration"
            const val COLUMN_NAME_DATE = "date"
        }
    }

    private class DBHelperRoutes internal constructor(context: Context) : SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {

        override fun onCreate(db: SQLiteDatabase) {
            db.execSQL(SQL_CREATE_ENTRIES)
        }

        override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
            db.execSQL(SQL_DELETE_ENTRIES)
            onCreate(db)
        }

        override fun onDowngrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
            onUpgrade(db, oldVersion, newVersion)
        }

        companion object {
            // METADATA
            private const val DATABASE_VERSION = 1
            private const val DATABASE_NAME = "Routes.db"
            // QUERIES
            private const val TEXT_TYPE = " TEXT"
            private const val REAL_TYPE = " REAL"
            private const val COMMA_SEP = ","
            private const val SQL_DELETE_ENTRIES = "DROP TABLE IF EXISTS " + RouteEntry.TABLE_NAME
            private const val SQL_CREATE_ENTRIES = "CREATE TABLE " + RouteEntry.TABLE_NAME + " (" +
                    RouteEntry.ID + " INTEGER PRIMARY KEY," +
                    RouteEntry.COLUMN_NAME_POS + REAL_TYPE + COMMA_SEP +
                    RouteEntry.COLUMN_NAME_NAME + TEXT_TYPE + COMMA_SEP +
                    RouteEntry.COLUMN_NAME_START_LAT + REAL_TYPE + COMMA_SEP +
                    RouteEntry.COLUMN_NAME_START_LNG + REAL_TYPE + COMMA_SEP +
                    RouteEntry.COLUMN_NAME_END_LAT + REAL_TYPE + COMMA_SEP +
                    RouteEntry.COLUMN_NAME_END_LNG + REAL_TYPE + COMMA_SEP +
                    RouteEntry.COLUMN_NAME_START_NAME + TEXT_TYPE + COMMA_SEP +
                    RouteEntry.COLUMN_NAME_END_NAME + TEXT_TYPE + " )"
        }
    }

    private class RouteEntry : BaseColumns {
        companion object {
            const val ID = BaseColumns._ID
            const val TABLE_NAME = "routes"
            const val COLUMN_NAME_POS = "pos"
            const val COLUMN_NAME_NAME = "name"
            const val COLUMN_NAME_START_LAT = "start_lat"
            const val COLUMN_NAME_START_LNG = "start_lng"
            const val COLUMN_NAME_END_LAT = "end_lat"
            const val COLUMN_NAME_END_LNG = "end_lng"
            const val COLUMN_NAME_START_NAME = "start_name"
            const val COLUMN_NAME_END_NAME = "end_name"
        }
    }
}



